packages/core/src/fetchers/fetchAllEntities.ts

```diff
@@ -0,0 +1,96 @@
+import { ContentfulClientApi, Entry } from 'contentful';
+import { MinimalEntryCollection } from './gatherAutoFetchedReferentsFromIncludes';
+
+const MIN_FETCH_LIMIT = 1;
+
+const fetchEntities = async ({
+  entityType,
+  client,
+  query,
+}: {
+  entityType: 'Entry' | 'Asset';
+  client: ContentfulClientApi<undefined>;
+  query: { 'sys.id[in]': string[]; locale: string; limit: number; skip: number };
+}) => {
+  if (entityType === 'Asset') {
+    return client.getAssets({ ...query });
+  }
+
+  return client.withoutLinkResolution.getEntries({ ...query });
+};
+
+export const fetchAllEntities = async ({
+  client,
+  entityType,
+  ids,
+  locale,
+  skip = 0,
+  limit = 100,
+  responseItems = [],
+}: {
+  client: ContentfulClientApi<undefined>;
+  entityType: 'Entry' | 'Asset';
+  ids: string[];
+  locale: string;
+  skip?: number;
+  limit?: number;
+  responseItems?: Entry[];
+}) => {
+  try {
+    if (!ids.length || !client) {
+      return {
+        items: [],
+        ...(entityType === 'Entry' && { includes: [] }),
```

Comment:
Wouldn't it try to `...(false)` if `entityType === 'Asset'`?

Reply:
The check will be false so it won't spread the includes object

Reply:
Apparently in JS the case I was worried about is handled. 🤔 I thought that attempting to spread an boolean into an object would result in an error, but looks like it's not the case

<img width="375" alt="Screenshot 2024-03-25 at 12 36 38" src="https://github.com/contentful/experience-builder/assets/4171202/0b025953-54fd-48d9-8c99-19d20c92a0dd">